toolkit/tools/pkg/specreaderutils/specreaderutil.go

```diff
@@ -704,6 +713,49 @@ func readBuildRequires(specFile, sourceDir, arch string, defines map[string]stri
 	return
 }
 
+func querySpecForTestFailureExpectation(specFile string) (result bool, err error) {
```

Comment:
IMHO it is little over complication. If the spec writer knows that the PTest is expected to fail, they can simply disable the check section. So, instead of imposing everyone to add the special comment, we can ask them to disable the check.

Reply:
Right now we find ourselves with a large number of failing tests. We're going to need to make a bulk change to baseline those known-failing tests. I'm definitely open to other approaches for capturing it if you have alternate recommendations.

And note I intentionally *don't* want to disable the check section. If a test is failing, IMHO we should still run it but not report the failure as blocking.